crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -273,6 +274,34 @@ pub enum EnvironmentError2 {
     BuildEnvCall(std::io::Error),
     #[error("error building environment: {0}")]
     BuildEnv(String),
+    #[error("provided lockfile path doesn't exist: {0}")]
+    BadLockfilePath(std::io::Error),
+    #[error("call to pkgdb failed: {0}")]
+    PkgDbCall(std::io::Error),
+    #[error("couldn't parse lockfile as JSON: {0}")]
+    ParseLockfileJSON(serde_json::Error),
+    #[error("couldn't write new lockfile contents: {0}")]
+    WriteLockfile(std::io::Error),
+    #[error("registry resolution failed: {0}")]
+    FailedResolution(PkgDbError),
+}
+
+/// A struct representing error messages coming from pkgdb
+#[derive(Debug, Deserialize)]
+pub struct PkgDbError {
+    /// The exit code of pkgdb, can be used to programmatically determine
+    /// the category of error.
+    pub exit_code: u64,
+    /// The generic message for this category of error.
+    pub category_message: String,
+    /// The more contextual message for the specific error that occurred.
+    pub context_message: String,
```

Comment:
suggestion blocking: match JSON format coming from pkgdb or add a comment that we're ignoring caught_message https://github.com/flox/pkgdb/blob/aec2f6ece0e5ed6c8f7e9e7e5b798a051d3e3db0/src/exceptions.cc#L34